homeassistant/components/elmax/config_flow.py

```diff
@@ -170,82 +177,65 @@ async def async_step_panels(
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])  # type: ignore[assignment]
         return await self.async_step_reauth_confirm()
 
-    async def async_step_reauth_confirm(self, user_input=None):
+    async def async_step_reauth_confirm(self, user_input=None) -> FlowResult:
         """Handle reauthorization flow."""
         errors = {}
         if user_input is not None:
-            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
-            password = user_input.get(CONF_ELMAX_PASSWORD)
-            entry = await self.async_set_unique_id(self._reauth_panelid)
+            username = user_input[CONF_ELMAX_USERNAME]
+            password = user_input[CONF_ELMAX_PASSWORD]
+            panel_pin = user_input[CONF_ELMAX_PANEL_PIN]
 
             # Handle authentication, make sure the panel we are re-authenticating against is listed among results
             # and verify its pin is correct.
+            reauth_panel_id = self._entry.data[CONF_ELMAX_PANEL_ID]
             try:
                 # Test login.
-                client = await self._async_login(
-                    username=self._reauth_username, password=password
-                )
-
+                client = await self._async_login(username=username, password=password)
                 # Make sure the panel we are authenticating to is still available.
                 panels = [
                     p
                     for p in await client.list_control_panels()
-                    if p.hash == self._reauth_panelid
+                    if p.hash == self._entry.data[CONF_ELMAX_PANEL_ID]
                 ]
                 if len(panels) < 1:
                     raise NoOnlinePanelsError()
 
-                # Verify the pin is still valid.from
+                # Verify the pin is still valid.
                 await client.get_panel_status(
-                    control_panel_id=self._reauth_panelid, pin=panel_pin
+                    control_panel_id=self._entry.data[CONF_ELMAX_PANEL_ID],
+                    pin=panel_pin,
                 )
 
-                # If it is, proceed with configuration update.
-                self.hass.config_entries.async_update_entry(
-                    entry,
-                    data={
-                        CONF_ELMAX_PANEL_ID: self._reauth_panelid,
-                        CONF_ELMAX_PANEL_PIN: panel_pin,
-                        CONF_ELMAX_USERNAME: self._reauth_username,
-                        CONF_ELMAX_PASSWORD: password,
-                    },
-                )
-                await self.hass.config_entries.async_reload(entry.entry_id)
-                self._reauth_username = None
-                self._reauth_panelid = None
-                return self.async_abort(reason="reauth_successful")
-
             except ElmaxBadLoginError:
                 _LOGGER.error(
                     "Wrong credentials or failed login while re-authenticating"
                 )
```

Comment:
```suggestion
```
No need to log this

Reply:
Done in 59b5c7f6e6cbebd005e9b74b73da31fa5612cb94, thank you!